### PR TITLE
Update Taskgraph to version 5.1.0 in CI

### PR DESCRIPTION
**Pull request by ahal**
This version includes a new key that should help app-services workaround a hardcoded Python path in the Mac workers.

No tests or changelog needed as changes are CI-only.

### Pull Request checklist ###

- [x] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/cut-a-new-release.md) after merging.
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes a changelog entry in [CHANGES_UNRELEASED.md](../CHANGES_UNRELEASED.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [x] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

[Branch builds](https://github.com/mozilla/application-services/blob/main/docs/howtos/branch-builds.md): add `[firefox-android: branch-name]` to the PR title.

### CODE REVIEW

**Comment by ahal** [comment]
@bendk once this lands, you should be able to add something like:
```
run:
    run-task-command: ["/path/to/my/python", "run-task"]
```
to the impacted task to workaround [your issue](https://firefox-ci-tc.services.mozilla.com/tasks/Y81pK7i4RkqJ7i-FgjP3lA/runs/0). I'm not sure where the Python is located on your workers, but try using `/usr/local/bin/python3`. That's what we have set in Gecko at least.

I still hope to solve this "properly" in the future, such that we can always reliably detect a Python without specific tasks needing to specify it.. But for now this seemed like the easiest way forward.

File: taskcluster/requirements.txt [hunk 1]
```diff
@@ -1,8 +1,8 @@
 #
-# This file is autogenerated by pip-compile with python 3.6
+# This file is autogenerated by pip-compile with python 3.7
```
Comment (hneiva) [inline taskcluster/requirements.txt:2]: 👀  I'm assuming this is tested and intentional

**Review by hneiva — APPROVED** [review]

File: taskcluster/requirements.txt (hunk 1 above)
Comment (jcristau) [reply taskcluster/requirements.txt:2]: mozillareleases/taskgraph:decision-mobile-35a3e77038af77191b9eb65d822a2ee34839e44d176859a69c5086f60101947c (as referenced in .tc.yml) has python 3.8.10, so this should be ok.

Comment (ahal) [reply taskcluster/requirements.txt:2]: Taskgraph also doesn't even run with 3.6 anymore, it has some 3.7+ only syntax :)

**Pull request merged**